### PR TITLE
[20542] Fix CI version management

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR unifies the CI in all workflows to follow this behavior:

- Trigger CI in all pull requests which HEAD branch is set to this supported branch version
- Remove nightly events in supported branches, since they are only run in default / base branch (master).
- Include missing ignore files for the sanitizer event trigger.
- Fix the default branch in workflows as this supported version branch.

master related PR:
- #4480 

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Changes are ABI compatible. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- **N/A** Check CI results: changes do not issue any warning.
- **N/A** Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/sanitizer-tests.yaml
Comment (elianalf) [inline .github/workflows/sanitizer-tests.yaml:1]: [line 131](https://github.com/eProsima/Fast-DDS/pull/4483/files#diff-37fd8971cb880bd169034bc59986b197ec187bd7193ef6edfd92a85d5e40ba07R131) should be 2.10.x

**Review by elianalf — CHANGES_REQUESTED** [review]

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by MiguelCompany** [comment]
@JesusPoderoso Would you mind resolving the conflicts here ?

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@mergifyio rebase

**Pull request merged**